verifier-sdk/src/light_transaction.rs

```diff
@@ -533,43 +526,39 @@ impl<
 
         let transaction_merkle_tree = self.get_transaction_merkle_tree()?;
 
-        // check merkle tree
-        for (i, leaves) in self.input.leaves.iter().enumerate() {
-            let mut msg = Vec::new();
-
-            if self.input.encrypted_utxos.len() > i * 256 {
-                msg.append(&mut self.input.encrypted_utxos[i * 256..(i + 1) * 256].to_vec());
-            }
-
-            // check account integrities
-            insert_two_leaves_cpi(
-                self.input.ctx.program_id,
-                &self
-                    .input
-                    .ctx
-                    .accounts
-                    .get_program_merkle_tree()
-                    .to_account_info(),
-                &self.input.ctx.accounts.get_authority().to_account_info(),
-                &self.input.ctx.remaining_accounts[NR_NULLIFIERS + i].to_account_info(),
-                &transaction_merkle_tree,
-                &self
-                    .input
-                    .ctx
-                    .accounts
-                    .get_system_program()
-                    .to_account_info(),
-                &self
-                    .input
-                    .ctx
-                    .accounts
-                    .get_registered_verifier_pda()
-                    .to_account_info(),
-                leaves[0],
-                leaves[1],
-                msg,
-            )?;
-        }
+        #[cfg(feature = "atomic-transactions")]
+        let two_leaves_pdas = Vec::new();
```

Comment:
Not necessarily to be addressed in this PR, but...

WRT our discussion whether we should ditch vectors all together - would you say there is a reasonable limit of leaves per verifier call which we could enforce?

Reply:
hm, I think the max limit would be the instruction data we can pass in one cpi call.
But a solana instruction also for cpi requires it's instruction data to be a vector so we need to be careful that we don't always consume a lot of memory if we just use a big array here.
And yes different pr.